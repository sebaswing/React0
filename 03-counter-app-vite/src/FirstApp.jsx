// import { Fragment } from "react";
// const GetName = (a,b) => {
//     return a + b;
// }

import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle='No hay subtítulo',name='No hay título' }) => {

// if (!title) {
//     throw new Error("El título no existe");
// }

    // console.log(props);
    return (
        // <Fragment>
        //<>   es lo mismo que un Fragment
        <> 
            {/* <h1>{ GetName(2,3) }</h1> */}
            <h1 data-testid="test-title">{title}</h1>
            <p>{subtitle}</p>
            <p>{subtitle}</p>
            <p>{name}</p>
        </>
        // </Fragment>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
}

// FirstApp.defaultProps = { deprectado
//     name: 'Sebastián Beltrán', // Propiedad por defecto,
//     subtitle: 'No hay subtítulo',
//     // title: 'No hay título',
// }
