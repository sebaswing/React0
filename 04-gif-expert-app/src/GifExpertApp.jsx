export const GifExpertApp = () => {
  return (
    <>
    {/*titulo  */}
      <h1>GifExpertApp</h1>

      {/*input */} 


      {/*listado de gifs */} 
       {/*gif item */}

    </>
  )
}
