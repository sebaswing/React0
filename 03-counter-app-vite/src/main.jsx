import React from "react";
import ReactDOM from "react-dom/client";
// import { HelloWorldApp } from "./HelloWorldApp";
import { FirstApp } from "./FirstApp";
import "./styles.css";
import { CounterApp } from "./CounterApp";



ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
        {/* <HelloWorldApp /> */}
        {/* <FirstApp title='Hola, Soy SebastiÃ¡n' /> */}
        <CounterApp value= { 20 } />
    </React.StrictMode>
)