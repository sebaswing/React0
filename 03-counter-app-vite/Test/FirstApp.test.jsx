import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp />', () => {

    // test('Debe hacer match con el snapshot', () => { 
        
    //     const  title = 'hola, Soy Gokú';
    //     const {container} = render(<FirstApp title = {title}/>);

    //     expect( container ).toMatchSnapshot();

    //  });

     test('Debe hacer match con el h1', () => { 
        
        const  title = 'hola, Soy Gokú';
        const {container,getByText,getByTestId} = render(<FirstApp title = {title}/>);

        expect( getByText(title) ).toBeTruthy;

        // const h1 = container.querySelector('h1');
        // // console.log(h1.innerHTML);
        // expect( h1.innerHTML ).toContain(title);

        expect( getByTestId('test-title').innerHTML).toBe(title);
     });

     test('Debe mostrar el subtítulo enviado por props', () => { 
        
        const  title = 'hola, Soy Gokú';
        const  subtitle = 'Soy un subtítulo';
        const {getAllByText} = render(
            <FirstApp 
                title = {title}
                subtitle = {subtitle}
            />
        );

        expect( getAllByText(subtitle).length ).toBe(2);
    });
});