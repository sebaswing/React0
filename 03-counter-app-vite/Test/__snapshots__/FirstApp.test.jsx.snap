// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    hola, Soy Gokú
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay título
  </p>
</div>
`;
